src/components/screens/monitor/delegates/delegates.test.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import Delegates from './delegates';
+import delegates from '../../../../../test/constants/delegates';
+
+const transformToLiskServiceFormat = ({ account, ...delegate }) => ({
+  ...delegate,
+  ...account,
+});
+const delegatesApiResponse = delegates.map(transformToLiskServiceFormat);
+
+jest.mock('../../../../constants/monitor', () => ({ DEFAULT_LIMIT: 8 }));
+
+describe('Delegates monitor page', () => {
+  let props;
+  let delegatesWithData;
+  const name = '1234';
+
+  const switchTab = (wrapper, tab) => {
+    wrapper.find(`.tab.${tab}`).simulate('click');
+    wrapper.setProps({
+      filters: {
+        ...props.filters,
+        tab,
+      },
+    });
+  };
+
+  beforeEach(() => {
+    props = {
+      t: key => key,
+      delegates: {
+        isLoading: true,
+        data: [],
+        loadData: jest.fn(),
+        clearData: jest.fn(),
+        urlSearchParams: {},
+      },
+      filters: {
+        tab: 'active',
+      },
+      applyFilters: jest.fn(),
+    };
+
+    delegatesWithData = {
+      ...props.delegates,
+      isLoading: false,
+      data: delegatesApiResponse,
+    };
+  });
+
+  it('renders a page with header', () => {
+    const wrapper = mount(<Delegates {...props} />);
+    expect(wrapper.find('Box header')).toIncludeText('Active delegates');
+  });
+
+  it('renders table with delegates', () => {
+    const wrapper = mount(<Delegates {...props} />);
+    expect(wrapper.find('.delegate-row')).toHaveLength(0);
+    wrapper.setProps({ delegates: delegatesWithData });
+    expect(wrapper.find('.delegate-row').hostNodes()).toHaveLength(delegatesApiResponse.length);
+  });
+
+  it('allows to switch to "Standby delegates" tab', () => {
+    const wrapper = mount(<Delegates {...{ ...props }} />);
+    expect(wrapper.find('.tab.standby')).not.toHaveClassName('active');
+    switchTab(wrapper, 'standby');
+    expect(wrapper.find('.tab.standby')).toHaveClassName('active');
+  });
+
+  it('shows error if API failed', () => {
+    const error = 'Loading failed';
+    const wrapper = mount(<Delegates {...props} />);
+    wrapper.setProps({
+      delegates: {
+        ...props.delegates,
+        isLoading: false,
+        error,
+      },
+    });
+    expect(wrapper).toIncludeText(error);
+  });
+
```

Comment:
Here you can add a setup function to mount the component and as you are using most of the time the same mount you can pass it in the beforeEach function.

something like.

const setup = (props) => mount(<Dekegates {...props}>);

beforeEach(() => wrapper = setup(props));

And in case you need to pass different props to specific test you can use setup fn